Parse/src/main/java/com/parse/ParseEncoder.java

```diff
@@ -120,6 +122,12 @@ public Object encode(Object object) {
         return ((ParseQuery.RelationConstraint) object).encode(this);
       }
 
+      if (object instanceof Pattern) {
+        JSONObject json = new JSONObject();
+        json.put("$regex", object.toString());
```

Comment:
Would it be possible to avoid `Pattern`s and move this logic to `ParseQuery` just like all other $ constraints?

I mean using `ParseQuery.addCondition(key, “$regex”, string)`. That way we keep consistencies and leave encoder untouched.